astropy/time/core.py

```diff
@@ -2228,32 +2280,93 @@ def to(self, unit, equivalencies=[]):
+          tm.to_value('jd', subfmt='decimal')
+          tm.to_value(format='jd', subfmt='decimal')
+
+        To convert to a unit with optional equivalencies, the options are::
+
+          tm.to_value('hr')  # convert to u.hr (hours)
+          tm.to_value('hr', [])  # specify equivalencies as a positional arg
+          tm.to_value('hr', equivalencies=[])
+          tm.to_value(unit='hr', equivalencies=[])
+
+        The built-in `~astropy.time.TimeDelta` options for ``format`` are:
+        {'jd', 'sec', 'datetime'}.
+
+        For the two numerical formats 'jd' and 'sec', the available ``subfmt``
+        options are: {'float', 'long', 'decimal', 'str', 'bytes'}. Here, 'long'
+        uses ``numpy.longdouble`` for somewhat enhanced precision (with the
+        enhancement depending on platform), and 'decimal' instances of
+        :class:`decimal.Decimal` for full precision.  For the 'str' and 'bytes'
+        sub-formats, the number of digits is also chosen such that time values
+        are represented accurately.  Default: as set by ``out_subfmt`` (which by
+        default picks the first available for a given format, i.e., 'float').
 
         Parameters
         ----------
+        format : str, optional
+            The format in which one wants the `~astropy.time.TimeDelta` values.
+            Default: the current format.
+        subfmt : str, optional
+            Possible sub-format in which the values should be given. Default: as
+            set by ``out_subfmt`` (which by default picks the first available
+            for a given format, i.e., 'float' or 'date_hms').
         unit : `~astropy.units.UnitBase` instance or str, optional
             The unit in which the value should be given.
         equivalencies : list of equivalence pairs, optional
             A list of equivalence pairs to try if the units are not directly
             convertible (see :ref:`unit_equivalencies`). If `None`, no
-            equivalencies will be applied at all, not even any set globally
-            or within a context.
+            equivalencies will be applied at all, not even any set globally or
+            within a context.
 
         Returns
         -------
         value : `~numpy.ndarray` or scalar
-            The value in the units specified.
+            The value in the format or units specified.
 
         See also
         --------
         to : Convert to a `~astropy.units.Quantity` instance in a given unit.
         value : The time value in the current format.
+
         """
+        # TODO: allow 'subfmt' also for units, keeping full precision
+        # (effectively, by doing the reverse of quantity_day_frac).
+        # This way, only equivalencies could lead to possible precision loss.
+        if ('format' in kwargs or
+                (args == () and 'unit' not in kwargs) or
+                (args != () and (args[0] is None or args[0] in self.FORMATS))):
```

Comment:
As above, following up on your previous comment, perhaps we should not allow `None` here (and possibly insist on at least 1 arg or kwarg)

Reply:
Agreed on both.